resp/get_foxsi_optics_effarea.pro

```diff
@@ -22,29 +24,31 @@ IF n_elements(offaxis_angle) EQ 1 THEN angle = 1/sqrt(2) * [offaxis_angle, offax
 ; Switch X0->D6 and X6->D0 - we do not need this now that we follow the optic number convention
 ;if MODULE_NUMBER eq 0 then MODULE = 6 else if MODULE_NUMBER eq 6 then MODULE = 0 $
 ;		else MODULE = MODULE_NUMBER
-		
-files =  GETENV('FOXSIPKG') + '/' + data_dir + 'FOXSI2_' + ['Module_X-' + num2str(MODULE_NUMBER) + '_EA_pan.txt', $
-         'Module_X-' + num2str(MODULE_NUMBER) + '_EA_tilt.txt']
 
-; todo: these need to be provided by the data files themselves
-angles = [-9, -7, -5, -3, -1, 0, 1, 3, 5, 7, 9]
-energy = [4.5,  5.5,  6.5,  7.5,  8.5,  9.5, 11. , 13. , 15. , 17. , 19. , 22.5, 27.5]
+
+CASE DATE OF
+    ; foxsi1
+    1.0679040e+09: files =  GETENV('FOXSIPKG') + '/' + data_dir + 'FOXSI2_' + ['Module_X-' + num2str(MODULE_NUMBER) + '_EA_pan.txt', $
+                       'Module_X-' + num2str(MODULE_NUMBER) + '_EA_tilt.txt'] ; for foxsi1 we used the same EA files as for foxsi2.    
+    ; foxsi2
+    1.1342592e+09: files =  GETENV('FOXSIPKG') + '/' + data_dir + 'FOXSI2_' + ['Module_X-' + num2str(MODULE_NUMBER) + '_EA_pan.txt', $
+                       'Module_X-' + num2str(MODULE_NUMBER) + '_EA_tilt.txt']
+    ; foxsi3
+    1.2522816e+09: files =  GETENV('FOXSIPKG') + '/' + data_dir + 'FOXSI3_' + ['Module_X-' + num2str(MODULE_NUMBER) + '_EA_pan.txt', $
+                       'Module_X-' + num2str(MODULE_NUMBER) + '_EA_tilt.txt']
+    ELSE: PRINT, 'DATE has an illegal value.'
```

Comment:
It's a good piece of advice! I change the CASE part of the code and instead, I included some IFs and ELSE to account for the scenarios you mentioned. I also added some warning prints in case the module numbers do not match with the flight. Something I wonder is the order that used for module+detector in FOXSI3. We put modules 8, 5, 4, 1, 0, 2 and 7 in front of positions 0, 1, 2, 3, 4, 5, and 6. Can this affect other parts of the foxsi-sciences codes?

Reply:
I think defining the module numbers in front of positions is the good way to do it, but we should be careful indeed about other pieces of code. I did not identify elsewhere where we have to make changes.
Another thing could be to code the module number for each position in the init_param...

Otherwise, I do not see the different IF and ELSE option for FOXSI3 here that you mentioned: did you commit or am I looking at the old version of the code?

Reply:
Here you will find the latest commit including the IF and ELSEs I mentioned: 

https://github.com/foxsi/foxsi-science/pull/85/commits/91168e77535b360e8e07014e78c22c0a1b858ef9

I think from the point of view of using the detectors part of FOXSI-soft we only use position numbers. If we stick to use positions instead of Det101 or CdT05  for example, we should be fine.